### PR TITLE
MEPTS-249:  implementation of  program transformation

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

File: src/main/resources/pdiresources/transformations/program-drop-temp-table.ktr [hunk 1]
```diff
@@ -0,0 +1,539 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nthfloor) [inline src/main/resources/pdiresources/transformations/program-drop-temp-table.ktr]: Do we want to "drop program" table as soon as the migration is complete?  What if we need to recover something after the migration is complete?

@psmatsinhe

**Review by nthfloor — CHANGES_REQUESTED** [review]
@cfaife 
I have updated the folder structure of hte PDI resource but in so doing I discovered a few things that need to be changed. They are not all related to this ticket, but I am unable to test the ticket till these get fixed.
1. Still referencing merge-patient-job on[ this line](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L47) even though that job is not in use.
2. There is this function `executeSetupCommand()` which us [hard coding](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L118-L167) pdi resources. It is not good design to do this because now we need to recomplile everytime we add a new resource. 
3. MySQLProps is not a service, please refactor this so that we are using the SettingsService for managing and access our mysql connection. The SettignsService can in turn use MySQLProps as a model class or something like that, but categorised in the correct directory.
4. [This line here](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L154-L156): what if the user has supplied the connection string via the settings file, do they need to add it again via the console? Same [here](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L166)...doesn't look like we are giving the user an option where to suppliy settings.
5. @psmatsinhe Have these transformations been tested through the app before? [This looks](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L106-L116) like it is using the merge-patient job and not the control-center job.
6. [This line](https://github.com/esaude/openmrs-database-migrator/blob/fa2942ec72b2154202d365e144b5c802cfc7dcdd/src/main/java/com/openmrs/migrator/Migrator.java#L149) I don't see this plugin in the code base - where is it coming from and why we copying it?

File: src/main/resources/pdiresources/transformations/program-drop-temp-table.ktr (hunk 1 above)
Comment (cfaife) [reply src/main/resources/pdiresources/transformations/program-drop-temp-table.ktr]: Hi Nathan all  change  requests are done

**Comment by cfaife** [comment]
Good Morning @nthfloor , this is dummy file used to allow the plugin folder to  be created. there is jar  plugin  generated  at runtime,

**Review by ynurmahomed — APPROVED** [review]

**Pull request merged**